### PR TITLE
Account transactions with filter

### PR DESCRIPTION
**Pull request by MarcusNkadimeng**
[Merge] feature/tracker → Develop
[Feature] - Account Transactions Screen
 
 ## Description 
Implemented the account transactions feature which has a filter feature using segmented control. It filters transactions by expenses and income. I also deleted the Transactions tab, it will now be the Tracker tab.
 
## Screenshots/GIFs 
Screen Recording 2024-05-14 at 00 37 17
 
## Checklist 
Please review and check the following before submitting this pull request: 
- [x] The code follows the project's coding standards. 
- [ ] 100% Unit test coverage for View Models 
- [x] All warnings Resolved.

### CODE REVIEW

File: budgetApp1/budgetApp1/AccountTransactionsViewController.swift [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+import UIKit
+
+class AccountTransactionsViewController: UIViewController {
+
+    // MARK: - UISpecs Dependency
+    private let uiSpecs = UISpecs()
+    
+    // MARK: - Variables
+    @IBOutlet weak var tableView: UITableView!
```
Comment (kieran-woodrow) [inline budgetApp1/budgetApp1/AccountTransactionsViewController.swift]: all of these private please

File: budgetApp1/budgetApp1/AccountTransactionsViewController.swift [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    @IBOutlet weak var segmentedControl: UISegmentedControl!
+    @IBOutlet weak var accountNameLabel: UILabel!
+    var selectedAccount: Account?
+    
+    private lazy var viewModel: AccountTransactionsViewModel = {
```
Comment (kieran-woodrow) [inline budgetApp1/budgetApp1/AccountTransactionsViewController.swift]: this is really strange.
Why can you not just do private lazy var viewModel = AccountTransactionsViewModel(repository: repository, delegate: self, accountID: selectedAccount.id)?
The viewModel relly shouldn't be so coupled to selectedAccount.id.
Is there no way around this?

File: budgetApp1/budgetApp1/AccountTransactionsViewController.swift [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        return AccountTransactionsViewModel(repository: repository, delegate: self, accountID: selectedAccount.id)
+    }()
+
+    // MARK: - functions
+    private func setupTableView() {
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.register(TransactionTableViewCell.nib(), forCellReuseIdentifier: NibIdentifiers.TransactionViewCellIdentifier)
+        tableView.contentInset = UIEdgeInsets(top: 0, left: 0, bottom: 1.0, right: 0)
+        viewModel.fetchTransactionsForAccount(accountID: selectedAccount?.id ?? "")
```
Comment (kieran-woodrow) [inline budgetApp1/budgetApp1/AccountTransactionsViewController.swift]: does this function belong in setUpTbleView?

Comment (kieran-woodrow) [reply budgetApp1/budgetApp1/AccountTransactionsViewController.swift]: please put all public functions first, then all private functions

File: budgetApp1/budgetApp1/AccountTransactionsViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+// MARK: - View Model
+class AccountTransactionsViewModel {
+    
+    // MARK: - Variables
+    private var repository: TransactionRepositoryType?
+    private weak var delegate: AccountsTransactionViewModelDelegate?
+    private var transactionList: [Transaction]?
+    internal var expenses: [Transaction]?
```
Comment (kieran-woodrow) [inline budgetApp1/budgetApp1/AccountTransactionsViewModel.swift]: no need for internal keyword.

File: budgetApp1/budgetApp1/TrackerViewController.swift [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+import UIKit
+
+class TrackerViewController: UIViewController {
+
+    // MARK: - UISpecs Dependency
+    private let uiSpecs = UISpecs()
+    
+    // MARK: - variables
+    @IBOutlet weak var tableView: UITableView!
```
Comment (kieran-woodrow) [inline budgetApp1/budgetApp1/TrackerViewController.swift]: private

Comment (kieran-woodrow) [reply budgetApp1/budgetApp1/TrackerViewController.swift]: is this file meant to be empty?

File: budgetApp1/budgetApp1/TrackerViewModel.swift [hunk 6]
```diff
@@ -5,29 +5,32 @@
 //  Created by Lehlohonolo Nkadimeng on 2024/04/14.
 //
 
-import Foundation
+import UIKit
 
-protocol TransactionViewModelDelegate: AnyObject {
+protocol TrackerViewModelDelegate: AnyObject {
```
Comment (kieran-woodrow) [inline budgetApp1/budgetApp1/TrackerViewModel.swift]: please rename this to TrackerViewModelProtocol

**Review by kieran-woodrow — CHANGES_REQUESTED** [review]
Just what i saw from the screenshot. Some numbers have a decimal value up to 1 decimal place and others have a decimal value up to 2 decimal places

File: budgetApp1/budgetApp1/TrackerViewController.swift (hunk 5 above)
Comment (MarcusNkadimeng) [reply budgetApp1/budgetApp1/TrackerViewController.swift]: Well this used to be the TransactionsViewController, I will be using this file in another feature hence it is still empty.

File: budgetApp1/budgetApp1/AccountTransactionsViewController.swift (hunk 2 above)
Comment (MarcusNkadimeng) [reply budgetApp1/budgetApp1/AccountTransactionsViewController.swift]: There is a better way around this. It has been refactored.

File: budgetApp1/budgetApp1/AccountTransactionsViewController.swift (hunk 3 above)
Comment (MarcusNkadimeng) [reply budgetApp1/budgetApp1/AccountTransactionsViewController.swift]: Not entirely, I moved it to the ViewDidLoad().

**Review by kieran-woodrow — APPROVED** [review]

**Review by AShozi — APPROVED** [review]

**Pull request merged**